lib/components/header/index.tsx

```diff
@@ -1,15 +1,50 @@
-import { Heading, Box } from '@chakra-ui/react';
+import { Heading, Box, Button, Text, Flex, useBreakpoint } from '@chakra-ui/react';
+import { PAGE_SIZES } from '../../constants';
 
 interface HeaderProps {
   header: String;
   style?: React.CSSProperties;
+  hasButton?: boolean;
+  buttonText?: String;
+  onButtonClick?: () => void;
 }
 
-const Header: React.FC<HeaderProps> = ({ header, style }: HeaderProps) => (
-  <Box style={style}>
-    <Heading mb={'3px'}>{header}</Heading>
-    <Box h={'4px'} bg={'var(--magenta)'} />
-  </Box>
-);
+const Header: React.FC<HeaderProps> = ({
+  header,
+  style,
+  hasButton,
+  buttonText,
+  onButtonClick,
+}: HeaderProps) => {
+  const breakpoint = useBreakpoint();
+
+  return (
+    <Box style={style}>
+      <Flex flexDir={'row'} justifyContent="space-between" mb={'3px'}>
+        <Heading>{header}</Heading>
+        {hasButton ? (
+          <Button
```

Comment:
Perhaps a separate functional component would be helpful here so you can name this button, because right now I'd have to dig where this is called to figure out the context/meaning of the button

Reply:
+1 to this!